driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala

```diff
@@ -172,11 +172,14 @@ object SparktaJob {
     apConfig.rollups.map(r => {
       val components = r.dimensionAndBucketTypes.map(dab => {
         dimensionsMap.get(dab.dimensionName) match {
-          case Some(x: Dimension) => x.bucketTypes.contains(new BucketType(dab.bucketType)) match {
-            case true => DimensionBucket(x, new BucketType(dab.bucketType, dab.configuration.getOrElse(Map())))
-            case _ =>
-              throw new DriverException(
-                "Bucket type " + dab.bucketType + " not supported in dimension " + dab.dimensionName)
+          case Some(x: Dimension) => if (x.bucketTypes.contains(dab.bucketType)) {
```

Comment:
Could you extract this if-else block to a function? This will improve readability